lib/aws-dev-utils/next_token_wrapper.rb

```diff
@@ -14,18 +14,23 @@ def method_missing m, *args, &block
     end
 
     private
-
-    def do_call m, *args, &block
-      @client.send(m , *args, &block).to_h.tap do |r|
+    #todo: add a condition for the route53 case and refactor var names
+    #irb -I lib of dev-utils
+    def do_call m, props = {}, &block
+      @client.send(m, props, &block).to_h.tap do |response|
         i = 1
-        key, key2 = extract_keys r
-        while(key && r[key] && i < @max) do
+        resp_keys, req_keys = extract_keys response
+
+        resp_keys = Array(resp_keys)
+        req_keys = Array(req_keys)
+
+        while(!resp_keys.empty? && resp_keys.all?{ |resp_key| response[resp_key] } && i < @max) do
```

Comment:
Maybe instead of `!resp_keys.empty?`, do an early exit if it is empty.
```ruby
next response if resp_keys.empty?
while(...) 
```

_I suggest `next` instead of `return`, because we are in a block._